src/Sylius/Component/Payment/Model/PaymentRequest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Model;
+
+use Sylius\Component\Resource\Model\TimestampableTrait;
+
+class PaymentRequest implements PaymentRequestInterface
+{
+    use TimestampableTrait;
+
+    protected ?string $hash = null;
+
+    protected ?PaymentMethodInterface $method = null;
+
+    protected ?PaymentInterface $payment = null;
```

Comment:
I agree, I did not remember one `sylius/sylius` entity requiring this, but the invoicing plugin have this kind of behaviour. I will change this to avoid making asserts when using the related getters.